lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -169,14 +169,23 @@ def authorize_update_app
   end
 
   def accepted_application_params
+    # We need to permit all because of dynamic data (and somehow params[cinstance] wont return `{plan_id}`)
+    params_permit = params.permit!
     # cinstance[*] naming is present for legacy reasons
-    application_attributes = params[:application] || params[:cinstance]
+    application_attributes = params_permit[:application] || params_permit[:cinstance]
     return {} unless application_attributes
+
     permitted_params = fields_definitions + %i[plan_id redirect_url]
-    application_attributes.permit(*permitted_params)
+    application_attributes.permit(*permitted_params).to_h
   end
 
   def fields_definitions
     FieldsDefinition.by_provider(site_account).by_target('Cinstance').pluck(:name)
   end
+
+  private
+
+  def local_params
+    params.permit(%i[plan_id service_id page]).to_h
```

Comment:
```suggestion
    params.permit(%i[plan_id service_id page])
```

In file parameters seem to be used as parameters, not hash. So perhaps remove `to_h`.